### ISSUE TITLE
No way to make non-syntactic changes visible

### ISSUE THREAD

**Issue by bmarwell** [Wilfred/difftastic#637]
**(1) A description of the issue. A screenshot is often helpful too.**

When there's only a non-syntactic diff (eg whitespace in perl, java, python source code), difft shows `no syntactic changes`.

However, for usage with `git diff`, I would like to see even those changes.

**(2) A copy of what you're diffing.**

n/a

(3) The version of difftastic you're using (see `difft --version`) and
your operating system.

```
$ difft -V
Difftastic 0.55.0 (built with rustc 1.75.0)
```

Running on MacOS, aarch64

**Comment by Wilfred** [comment]
I agree this is sometimes useful, but it's quite hard to do with an AST diff. It might fit into the patch mode I'm looking at right now, but overall this is lower priority right now I'm afraid.

**Comment by bmarwell** [comment]
Thanks, and no worries... just consider it a request for enhanced.

**Comment by ctxnop** [comment]
I also need this, as a reviewer I need to see diffs that are only style, because I want to spot a style deviation from coding rules, or trailing spaces, or mix or tab/spaces for indentations.

I do understand that it might be hard to do with an AST, but I would argue that it can be a 3 steps job:
- first do the AST aware diff
- then do a normal text diff (you might just use the venerable diff tool here)
- both tools outputs as chunks, we should be able to merge both outputs. If a chunk is covered by both, just ouput the AST aware one.

**Comment by ndevenish** [comment]
Switched to difftastic for showing diffs a couple of days ago.... reverting back because a diff tool that doesn't show some of the diff is useless for me.

Maybe this should be listed as a known issue or non-goal, or at least a warning at the top of the git instructions page. It would have saved me some time, and I didn't parse "a structural diff that understands syntax" as "a structural diff that ONLY understands syntax".

**Comment by bmarwell** [comment]
I just use git difftool for most changes, git diff for whitespace changes - which is done rarely b/c of code formatters

**Comment by matijs** [comment]
I have `difft` aliased to `-c diff.external=difft diff`. This is just 1 character more than `diff` but has the advantage of keeping `git diff` around for `git diff --ws-error-highlight=all`.

**Comment by miaLDN** [comment]
I switched git to difft recently, but going back to git-diff or delta.

I'm snowed under with difft showing white space changes; OR even worse the entire file due to EOL changing from local to repo standard on commit (and vice versa on checkout).

I just checked all the options and I see noting about including or not excluding white space.
